src/hooks/useReferralAddress.ts

```diff
@@ -9,16 +11,21 @@ export const useReferralAddress = (refCode: string) => {
     if (!compositeClient || !refCode) {
       return undefined;
     }
-    const endpoint = `${compositeClient.indexerClient.config.restEndpoint}/v4/affiliates/address`;
-    const response = await fetch(`${endpoint}?referralCode=${encodeURIComponent(refCode)}`, {
-      method: 'GET',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-    });
+    try {
+      const endpoint = `${compositeClient.indexerClient.config.restEndpoint}/v4/affiliates/address`;
```

Comment:
hmm you dont actually need to wrap contents of react-query queryFn inside a try/catch - it will already handle errors for you and is returned as part of the `error` property, seems more semantic for either each component to handle errors themselves or add middleware to react-query to add logging to all errors

Reply:
can do!